mobileClient/components/WalletsList.js

```diff
@@ -192,80 +183,79 @@ export default class WalletsList extends React.Component {
+    };
     return (
       <List containerStyle={{ borderTopWidth: 0, borderBottomWidth: 0 }}>
-        <SectionList
-          sections={[
-            {
-              data: this.state.searchResults,
-              key: "Add a Wallet",
-              keyExtractor: item => item.id,
-              renderItem: ({ item }) => {
-                const coinAvatarUrls = {
-                  btc:
-                    "https://cdn3.iconfinder.com/data/icons/inficons-currency-set/512/btc-512.png",
-                  btc_test:
-                    "http://bitcoinist.com/wp-content/themes/bitcoinist/img/Bitcoin-price-icon.png",
-                  eth:
-                    "https://cdn4.iconfinder.com/data/icons/cryptocoins/227/ETH-512.png",
-                  eth_test:
-                    "https://cdn4.iconfinder.com/data/icons/cryptocoins/227/ETH-512.png"
-                };
-                return (
-                  <ListItem
-                    roundAvatar
-                    title={`${item.coin}`}
-                    avatar={{ uri: coinAvatarUrls[item.coinAbbr] }}
-                    containerStyle={{ borderBottomWidth: 0 }}
-                    rightTitle="Add Wallet"
-                    rightIcon={{ name: "add" }}
-                    onPressRightIcon={() => {
-                      this.addWallet(item.coinAbbr);
-                    }}
-                  />
-                );
-              }
-            },
-            {
-              data: this.state.wallets,
-              key: "Your Wallets",
-              keyExtractor: item => item.coinAbbr,
-              renderItem: ({ item }) => {
-                const coinAvatarUrls = {
-                  btc:
-                    "https://cdn3.iconfinder.com/data/icons/inficons-currency-set/512/btc-512.png",
-                  btc_test:
-                    "http://bitcoinist.com/wp-content/themes/bitcoinist/img/Bitcoin-price-icon.png",
-                  eth:
-                    "https://cdn4.iconfinder.com/data/icons/cryptocoins/227/ETH-512.png",
-                  eth_test:
-                    "https://cdn4.iconfinder.com/data/icons/cryptocoins/227/ETH-512.png"
-                };
-                return (
-                  <ListItem
-                    roundAvatar
-                    title={`${item.coin}`}
-                    subtitle={`${item.balance}`}
-                    avatar={{ uri: coinAvatarUrls[item.coinAbbr] }}
-                    containerStyle={{ borderBottomWidth: 0 }}
-                  />
-                );
+        {this.state.searchResults && this.state.wallets ? (
+          <SectionList
+            ItemSeparatorComponent={this.renderSeparator}
+            SectionSeparatorComponent={this.renderSectionSeparator}
+            ListHeaderComponent={this.renderHeader}
+            ListFooterComponent={this.renderFooter}
+            onRefresh={this.handleRefresh}
+            refreshing={this.state.refreshing}
+            extraData={this.state}
```

Comment:
what does extraData do?

Reply:
extraData tells the component where to look for changes that should cause a re-render. SectionLists and FlatLists apparently won't re-render if their data sources don't have major changes, setting extraData gives it something else to look at to trigger a re-render.

For this component though, I don't know if it's actually doing anything. I needed to use it on the FriendsList to make it re-render correctly when adding friends. I just copied all of that over to the WalletList. I might try removing it to see if it has any impact.